service_container/tags.rst

```diff
@@ -564,3 +564,241 @@ application handlers.
+
+In the following example, all services tagged with app.handler are passed as
+first constructor argument to the App\HandlerCollection service, like previously,
+but we can now access to a specific injected service :
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            App\Handler\One:
+                tags:
+                    - { tag: 'app.handler', index_attribute_name : 'hOne' }
+
+            App\Handler\Two:
+                tags:
+                    - { tag: 'app.handler', index_attribute_name: 'hTwo' }
+
+            App\HandlerCollection:
+                # inject all services tagged with app.handler as first argument
+                arguments: [!tagged { tag: 'app.handler', index_by: 'index_attribute_name' }]
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="App\Handler\One">
+                    <tag name="app.handler" index_attribute_name="hOne" />
+                </service>
+
+                <service id="App\Handler\Two">
+                    <tag name="app.handler" index_attribute_name="hTwo" />
+                </service>
+
+                <service id="App\HandlerCollection">
+                    <!-- inject all services tagged with app.handler as first argument -->
+                    <argument type="tagged" tag="app.handler" index-by="index_attribute_name"/>
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+        use Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument;
+
+        $container->register(App\Handler\One::class)
+            ->addTag('app.handler', ['index_attribute_name' => 'hOne']);
+
+        $container->register(App\Handler\Two::class)
+            ->addTag('app.handler', ['index_attribute_name' => 'hTwo']);
+
+        $container->register(App\HandlerCollection::class)
+            // inject all services tagged with app.handler as first argument
+            ->addArgument(new TaggedIteratorArgument('app.handler', 'index_attribute_name'));
+
+After compilation the ``HandlerCollection`` service is able to iterate over your
+application handlers. But we want to retrieve a specific service by it's
+``index_attribute_name``. As it's an iterator, we can use ``iterator_to_array``
+to get an array and then retrieve the ``hTwo`` handler :
+
+.. code-block:: php
+
+    // src/HandlerCollection.php
+    namespace App;
+
+    class HandlerCollection
+    {
+        public function __construct(iterable $handlersIt)
```

Comment:
Would rename it to „$handlers“ here

Also „hOne“ should be „handler_one“ as it’s a key and more speakable